Yank/sampling.py

```diff
@@ -789,7 +789,10 @@ def _initialize_netcdf(self):
 
     def _write_iteration_netcdf(self):
         super(ModifiedHamiltonianExchange, self)._write_iteration_netcdf()
-        if self.fully_interacting_state is not None:
+        if self.mpicomm:
+            # Only the root node will write data.
+            if self.mpicomm.rank != 0: return
```

Comment:
Agreed! We should probably protect master and switch to using feature branches instead